backend/core/views/game.py

```diff
@@ -100,16 +104,17 @@ def save_pad(request, player, uuid):
         PlayerFinishedMessage(pad.initial_player).serialize(),
     )
 
-    with transaction.atomic():
-        game = Game.objects.select_for_update().get(uuid=game.uuid)
-        game.pads_done = game.pads_done + 1
-        game.save()
+    if should_increment_game_pads_done:
```

Comment:
en fait cette manière de faire ça dénormalise la bdd (on a un champ pads_done qui dépend de données qui existent sont dans la table pad)
Est-ce que tu peux pas plutôt à cet endroit regarder len([pad for pad in game.pads if pad.sentence is not null]) et supprimer le champ pads_done ?
Le désavantage c'est la perf mais vu que c'est que au niveau de la game ça va seulement faire une requête en plus donc je pense pas que ça va beaucoup jouer.
Et ça permet aussi d'enlever la logique un peu bizarre de should_increment_game_pads_done

Reply:
C'était ce que j'avais codé au début, mais ça marchait pas avec le locking : j'arrivais pas à "locker" un ensemble de pad steps. Beaucoup plus simple de locker la game, incrémenter un compteur et délocker la game (ça c'est le cas d'école du locking de bdd donc hyper bien documenté en django)